### DISCUSSION TITLE
[WASM/iOS] Canvas is lagging on iPhones and macs

### DISCUSSION THREAD

**Discussion by asadullah-refat-89** [unoplatform/uno#10609] [Q&A]
I have developed some games with Uno. The games target only the WebAssembly head. So far the performance is great on all browsers in Android and Windows. But the app is lagging a lot on iOS and mac. I have enabled 4GB memory too. The games are in Interpreter mode. There are 5 games in total. I would really appreciate it if there is anything that can be done to improve performance on iOS and mac. I have primarily used Canvases and timers to render everything and move the objects through the canvas.

Thanks in advance.

Links to the games:
https://selise-space-shooter.seliselocal.com/
https://selise-sky-racer.seliselocal.com/
https://selise-hungry-worm.seliselocal.com/
https://selise-memory-match.seliselocal.com/
https://selise-candy-craze.seliselocal.com/

**Comment by jeromelaban** [comment]
Thanks for the report! Are you using AOT compilation for WebAssembly?

**Reply by asadullah-refat-89** [reply]
Nope using Interpreter mode compilation.

**Reply by jeromelaban** [reply]
It may help to use AOT: https://platform.uno/docs/articles/external/uno.wasm.bootstrap/doc/runtime-execution-modes.html#mixed-interpreter-and-aot-mode

**Reply by asadullah-refat-89** [reply]
Alright, @jeromelaban I will try that and Update you. Another key piece of information is I am using .NET 7.
